src/Catalyst.Core.Modules.Keystore/KeyStoreService.cs

```diff
@@ -0,0 +1,809 @@
+                }
+                catch (Exception e)
+                {
+                    throw new UnauthorizedAccessException("The pass phrase is wrong.", e);
+                }
+            }
+
+            log.Debug("Pass phrase is okay");
+        }
+
+        /// <summary>
+        ///   Find a key by its name.
+        /// </summary>
+        /// <param name="name">
+        ///   The local name of the key.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result is
+        ///   an <see cref="Microsoft.EntityFrameworkCore.Metadata.IKey"/> or <b>null</b> if the the key is not defined.
+        /// </returns>
+        public async Task<IKey> FindKeyByNameAsync(string name, CancellationToken cancel = default)
+        {
+            var key = await Store.TryGetAsync(name, cancel).ConfigureAwait(false);
+            if (key == null)
+            {
+                return null;
+            }
+            
+            return new KeyInfo
+            {
+                Id = key.Id, Name = key.Name
+            };
+        }
+
+        /// <summary>
+        ///   Gets the IPFS encoded public key for the specified key.
+        /// </summary>
+        /// <param name="name">
+        ///   The local name of the key.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result is
+        ///   the IPFS encoded public key.
+        /// </returns>
+        /// <remarks>
+        ///   The IPFS public key is the base-64 encoding of a protobuf encoding containing 
+        ///   a type and the DER encoding of the PKCS Subject Public Key Info.
+        /// </remarks>
+        /// <seealso href="https://tools.ietf.org/html/rfc5280#section-4.1.2.7"/>
+        public async Task<string> GetPublicKeyAsync(string name, CancellationToken cancel = default)
+        {
+            // TODO: Rename to GetIpfsPublicKeyAsync
+            string result = null;
+            var ekey = await Store.TryGetAsync(name, cancel).ConfigureAwait(false);
+            if (ekey != null)
+            {
+                UseEncryptedKey(ekey, key =>
+                {
+                    var kp = GetKeyPairFromPrivateKey(key);
+                    var spki = SubjectPublicKeyInfoFactory.CreateSubjectPublicKeyInfo(kp.Public).GetDerEncoded();
+
+                    // Add protobuf cruft.
+                    var publicKey = new Lib.Cryptography.Proto.PublicKey
+                    {
+                        Data = spki
+                    };
+                    switch (kp.Public) 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)